src/status_im/protocol/listeners.cljs

```diff
@@ -3,28 +3,34 @@
             [status-im.protocol.ack :as ack]
             [status-im.protocol.web3.utils :as u]
             [status-im.protocol.encryption :as e]
-            [taoensso.timbre :refer-macros [debug]]
+            [taoensso.timbre :as log]
             [status-im.utils.hex :as i]))
 
+(defn create-error [description]
+  (log/debug :parse-payload-error description)
+  {:error description})
+
 (defn- parse-payload [payload]
-  (debug :parse-payload)
+  (log/debug :parse-payload)
   (try
     ;; todo figure why we have to call to-utf8 twice
-    (let [read (comp r/read-string u/to-utf8 u/to-utf8)]
-      {:payload (read payload)})
+    (let [parsed (-> payload u/to-utf8 u/to-utf8 r/read-string)]
+      (if (map? parsed)
+        {:payload parsed}
+        (create-error (str "Invalid payload type " (type parsed)))))
```

Comment:
actually this isn't perfect fix tbh. I haven't figured yet why, but this code has better chances to parse payload https://github.com/status-im/status-react/pull/1584/files#diff-a87a8f8fddf1875274bd71f107d97816R13